### ISSUE TITLE
Add before/after tests to stealth evasions

### ISSUE THREAD

**Issue by berstend** [berstend/puppeteer-extra#78]
As a maintainer it's really hard to otherwise merge in new code mods, as I don't know if they do what they advertise without side-effects :)

I plan to tackle this after #21

**Comment by berstend** [comment]
Also add specific iframe test to prevent regressions (#60)

**Comment by berstend** [comment]
Also we can codify tests against: https://github.com/antoinevastel/fpscanner

**Comment by shirshak55** [comment]
@berstend we need to go to distil network website that is the first verification for stealth tests?

**Comment by berstend** [comment]
@shirshak55 do they have a public test page to test against? can't seem to find one.

**Comment by shirshak55** [comment]
@berstend just go to https://www.distilnetworks.com/ . They usually show something like canvas error and redirect to captcha page .

**Comment by shirshak55** [comment]
basically they try to load something like font etc and as we are using patched canvas  or something their fingerprinting fails and redirect to captcha page.

BTW i have not tested with new stealth but soon will :)

**Comment by berstend** [comment]
@shirshak55 let me know how it goes :)

If you can find/make a demo page with any new tricks or problems I can make updates to the stealth plugin to fix them, without a demo page it's not really possible. :)

**Comment by shirshak55** [comment]
I think this issue is also related with this #28

**Comment by shirshak55** [comment]
hmm u can go to https://www.distilnetworks.com/ using puppeteer extra and u will notice error message on console.

**Comment by berstend** [comment]
`webgl.vendor` should fix the problem mentioned in #28, if it doesn't than that's a bug (but again, no code provided to repro)

**Comment by berstend** [comment]
Ah, it was this issue:

```
VM43:14 Uncaught TypeError: getParameter is not a function
    at WebGLRenderingContext.getParameter (<anonymous>:14:18)
    at a.getWebglFp (zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1)
    at a.webglKey (zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1)
    at a.interrogate (zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1)
    at zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1
```

This is fixed since yesterday, lol.

**Comment by berstend** [comment]
Just confirmed it myself, the distil homepage loads correctly with `puppeteer-extra-plugin-stealth@2.4.4` :)

**Comment by berstend** [comment]
The `webgl.vendor` was some copy pasta from the early days, I did my own proper implementation of that fix yesterday.

There are also some other "fixed" I want to do again in the future (like the console.debug one).

**Comment by shirshak55** [comment]
yea I am happy this issue was fix

```
VM43:14 Uncaught TypeError: getParameter is not a function
    at WebGLRenderingContext.getParameter (<anonymous>:14:18)
    at a.getWebglFp (zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1)
    at a.webglKey (zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1)
    at a.interrogate (zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1)
    at zhrodsadknkfnugjfcevfffxabysrrrbbd.js:1
```

Btw do you know this headless behave like this like why by default it don't have getParameter etc.. and we need to patch it ? Because it will always be like cat and rat issu.

**Comment by momala454** [comment]
> 
> 
> > Btw do you know this headless behave like this like why by default it don't have getParameter etc.. and we need to patch it ? Because it will always be like cat and rat issu.
> 
> That's due to certain things being attached to the Chrome UI code, which isn't being loaded in headless mode. :)
> 
> Closing this for now, as all relevant and new evasions have tests. A standalone html test page is a nice to have I keep in mind.

can you make an issue for the standalone tests or reopen this issue ?
